### PR TITLE
Doc improvements.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: lib.rs [hunk 1]
```diff
@@ -216,7 +224,7 @@ impl ThreadPool {
             name: name,
             job_receiver: Mutex::new(rx),
             empty_condvar: Condvar::new(),
-            empty_trigger: Mutex::new( () ),
+            empty_trigger: Mutex::new(()),
```
Comment (dns2utf8) [inline lib.rs]: I would have left that for readability

File: lib.rs [hunk 2]
```diff
@@ -235,14 +243,50 @@ impl ThreadPool {
...
+    /// Execute four jobs on a thread pool that can run two jobs concurrently:
+    ///
+    /// ```
+    /// use threadpool::ThreadPool;
+    ///
+    /// let pool = ThreadPool::new(2);
+    /// pool.execute(|| println("hello"));
+    /// pool.execute(|| println("world"));
+    /// pool.execute(|| println("foo"));
+    /// pool.execute(|| println("bar"));
```
Comment (dns2utf8) [inline lib.rs]: The `println` does not inwoke the macro. -> !

File: lib.rs [hunk 3]
```diff
@@ -235,14 +243,50 @@ impl ThreadPool {
...
+    /// use std::thread::sleep;
+    ///
+    /// let pool = ThreadPool::new(2);
+    /// for _ in 0..10 {
+    ///     pool.execute(|| {
+    ///         sleep(Duration::from_secs(100));
+    ///     });
+    /// }
+    ///
+    /// sleep(Duration::from_secs(1)); // wait for threads to start
```
Comment (dns2utf8) [inline lib.rs]: cool 👍

File: lib.rs [hunk 4]
```diff
@@ -284,15 +338,17 @@ impl ThreadPool {
...
     ///
-    /// let num_threads = 10;
-    /// let pool = ThreadPool::new(num_threads);
-    /// for _ in 0..num_threads {
+    /// let pool = ThreadPool::new(4);
+    /// for n in 0..10 {
     ///     pool.execute(move || {
-    ///         panic!()
+    ///         if n % 2 == 0 {
+    ///             panic!()
```
Comment (dns2utf8) [inline lib.rs]: Maybe add a comment like "simulating panic" so the test output does not look that grimm

File: lib.rs [hunk 5]
```diff
@@ -311,9 +367,45 @@ impl ThreadPool {
...
+    /// // Increase thread capacity of the pool
+    /// pool.set_num_threads(8);
+    ///
+    /// sleep(Duration::from_secs(1)); // wait for new threads to start
+    /// assert_eq!(8, pool.active_count());
+    /// assert_eq!(2, pool.queued_count());
+    ///
+    /// // Decrease thread capacity of the pool
+    /// // No active threads are killed
+    /// pool.set_num_threads(4);
```
Comment (dns2utf8) [inline lib.rs]: 👍

File: lib.rs [hunk 6]
```diff
@@ -311,9 +367,45 @@ impl ThreadPool {
...
+    ///
+    /// assert_eq!(8, pool.active_count());
+    /// assert_eq!(2, pool.queued_count());
+    /// ```
     pub fn set_num_threads(&mut self, num_threads: usize) {
         assert!(num_threads >= 1);
-        let prev_num_threads = self.shared_data.max_thread_count.swap(num_threads, Ordering::Release);
+        let prev_num_threads = self.shared_data.max_thread_count.swap(
```
Comment (dns2utf8) [inline lib.rs]: I wonder what rule you would like to implement for the linewrap. Is it the length of the line?

Comment (frewsxcv) [inline lib.rs]: I'm not sure, `cargo fmt` wrapped this and I'm not sure what its rules are

File: lib.rs (hunk 1 above)
Comment (frewsxcv) [inline lib.rs]: `cargo fmt` did this, but I can revert it

**Comment by dns2utf8** [comment]
Well if the rule is "what ever `cargo fmt` says" I think it is ok.

File: lib.rs [hunk 7]
```diff
@@ -322,8 +414,7 @@ impl ThreadPool {
         }
     }
 
-    /// Block the current thread until all jobs in the pool are completed.
-    /// Many threads can wait for a pool to finish concurrently.
+    /// Block the current thread until all jobs in the pool have been executed.
```
Comment (dns2utf8) [inline lib.rs]: I would add a text like this: 


A pool can be joined repeatedly. 

Calling the join method on an empty pool will cause it to return immediately.

The join method may be used from multiple threads at the same time.

# Warning

Joining the pool from within the pool will cause a deadlock.
This behavior is considered safe.

**Pull request merged**